src/server/utils/wallets/createSmartWallet.ts

```diff
@@ -0,0 +1,152 @@
+import { defineChain, type Address } from "thirdweb";
+import { smartWallet, type Account } from "thirdweb/wallets";
+import { createWalletDetails } from "../../../db/wallets/createWalletDetails";
+import { WalletType } from "../../../schema/wallet";
+import { thirdwebClient } from "../../../utils/sdk";
+import { splitAwsKmsArn } from "./awsKmsArn";
+import {
+  createAwsKmsWallet,
+  type CreateAwsKmsWalletParams,
+} from "./createAwsKmsWallet";
+import {
+  createGcpKmsWallet,
+  type CreateGcpKmsWalletParams,
+} from "./createGcpKmsWallet";
+import { createLocalWallet } from "./createLocalWallet";
+import { getAwsKmsAccount } from "./getAwsKmsAccount";
+import { getGcpKmsAccount } from "./getGcpKmsAccount";
+
+const createSmartWallet = async ({
+  adminAccount,
+  accountFactoryAddress,
+}: {
+  adminAccount: Account;
+  accountFactoryAddress?: Address;
+}) => {
+  const smartAccount = smartWallet({
+    chain: defineChain(1),
```

Comment:
I assume you're only using this to predict the smart wallet address?

The danger here is that it requires the account factory to be there on that chain.

I think for smart accounts it's better to recreate the smart wallet at runtime based on the chain coming in the request.

So you would create a normal backend wallet and just give it a smart flag in the db (which you kinda do already with the factory address) then on every request, compute the smart wallet based on the chain in the request. If factory not available on that chain = error.

Reply:
> I think for smart accounts it's better to recreate the smart wallet at runtime based on the chain coming in the request.

That's the intention here, did I miss something in the implementation?

> then on every request, compute the smart wallet based on the chain in the request
This would be very confusing. We must ensure smart wallet has same account address on every chain for UX. I assumed this was already always the case, so we are storing the predicted account address.

> If factory not available on that chain = error.
exactly